### PR TITLE
Plugin system 1.1

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Updated the IRIDA Plugin system to version 1.1.  This will require new plugins to use the service-based metadata system built in https://github.com/phac-nml/irida/pull/576.

This also includes documentation on how developers can update their plugins.

## Related issue
Related to IRIDA PR https://github.com/phac-nml/irida/pull/576.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
~* [ ] Tests added (or description of how to test) for any new features.~
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks great. Thanks @tom114 . Just two small changes.

File: doc/administrator/upgrades/index.md [hunk 1]
```diff
@@ -89,6 +89,65 @@ You can download this script at <https://github.com/phac-nml/irida/tree/developm
 $ perl metadata-mappings.pl -u [database username] -p [database password] -d [irida database name] -h [database host] > [outputfile.csv]
 ```
 
+## Plugin version updates
+
+Due to the updates described above regarding IRIDA's metadata storage refactors, the way that IRIDA analysis plugins interact with a sample's metadata will change as well.  If you are the developer of an IRIDA analysis plugin, you will need to perform the updates below and release a new plugin version for your pipeline to be compatible with new versions of IRIDA.  This guide makes some assumptions that your plugin is structured similarly to IRIDA's example plugin <https://github.com/phac-nml/irida-plugin-example>.
+
+There are 2 files that must be updated:
+
+* The plugin's `AnalysisSampleUpdater` implementation.  This will need to change any methods interacting with sample metadata to use the new service methods.
```
[the remark this answers is not in the document]
Comment (tom114) [reply doc/administrator/upgrades/index.md]: Thanks Aaron.  Updated in 1fd4abe6e2

File: doc/administrator/upgrades/index.md [hunk 2]
```diff
@@ -89,6 +89,65 @@ You can download this script at <https://github.com/phac-nml/irida/tree/developm
...
+* The `pom.xml` file.  This will tell your IRIDA plugin to target the `21.01` version of IRIDA's code in order to take advantage of the new features.  You will also need to increment your plugin version number in this file to tell IRIDA to use your new version.
```
[the remark this answers is not in the document]
Comment (tom114) [reply doc/administrator/upgrades/index.md]: Thanks Aaron.  Updated in 1fd4abe6e2

**Review by apetkau — APPROVED** [review]

**Comment by apetkau** [comment]
Looks good. Thanks.

**Pull request merged**